packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -58,15 +58,18 @@ class ConfigChainBuilder {
     let foundIgnore = false;
 
     while (loc !== (loc = path.dirname(loc))) {
+      // Find configs in the following file type precedence order:
+      // .babelrc -> .babelrc.js -> package.json
       if (!foundConfig) {
         const configLoc = path.join(loc, BABELRC_FILENAME);
-        if (exists(configLoc)) {
-          this.addConfig(configLoc);
-          foundConfig = true;
-        }
-
+        const configJSLoc = path.join(loc, BABELRC_JS_FILENAME);
         const pkgLoc = path.join(loc, PACKAGE_FILENAME);
-        if (!foundConfig && exists(pkgLoc)) {
+
+        if (exists(configLoc)) {
```

Comment:
The precedence is error prone. Since `.babelrc` is a hidden file is most cases, if you inadvertently have one in the same directory and you try to configure Babel using the `babelrc.js` file: your are going to have a bad time.

IMO you should throw an error in that case. I'm not sure for the configuration located in the `package.json`, should we also throw?

The migration between `.babelrc` and `babelrc.js` isn't difficult (maybe except for the `env` key), so I think it's to force users to update it.